python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -0,0 +1,200 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+import math
+from typing import Dict, Any, List, Tuple
+
+
+def principal_value(angle_in_radians: float) -> float:
+    """
+    Ensures the angle is within [-pi, pi).
+    :param angle_in_radians: Angle in radians.
+    :return: Scaled angle in radians.
+    """
+
+    interval_min = -math.pi
+    two_pi = 2 * math.pi
+    scaled_angle = (angle_in_radians - interval_min) % two_pi + interval_min
+    return scaled_angle
+
+
+def compute_segment_sign(arcline_path_3: Dict[str, Any]) -> Tuple[float, float, float]:
+    """
+    Compute the sign of an arcline_path_3 based on its shape.
+    :param arcline_path_3: arcline_patch_3 record.
+    :return: Tuple of signs for all three parts of the path. 0 if straight, -1 if right,
+        1 if left.
+    """
+    shape = arcline_path_3['shape']
+    segment_sign = [0, 0, 0]
+
+    if shape in ("LRL", "LSL", "LSR"):
+        segment_sign[0] = 1
+    else:
+        segment_sign[0] = -1
+
+    if shape == "RLR":
+        segment_sign[1] = 1
+    elif shape == "LRL":
+        segment_sign[1] = -1
+    else:
+        segment_sign[1] = 0
+
+    if shape in ("LRL", "LSL", "RSL"):
+        segment_sign[2] = 1
+    else:
+        segment_sign[2] = -1
+
+    return segment_sign
+
+
+def get_transformation_at_s(lie_group_pose: Tuple[float, float, float],
+                            s: float) -> Tuple[float, float, float]:
+    """
+    Get the affine transformation at s meters along the path.
+    :param lie_group_pose: Pose represented as tuple (x, y, yaw).
+    :param s: Length along the arcline path in range (0, length_of_arcline_path].
+    :return: Transformation represented as pose tuple.
+    """
+
+    theta = lie_group_pose[2] * s
+    ctheta = math.cos(theta)
+    stheta = math.sin(theta)
+
+    if abs(lie_group_pose[2]) < 1e-6:
+        return [lie_group_pose[0] * s, lie_group_pose[1] * s, theta]
+    else:
+        new_x = (lie_group_pose[1] * (ctheta - 1.0) + lie_group_pose[0] * stheta) / lie_group_pose[2]
+        new_y = (lie_group_pose[0] * (1.0 - ctheta) + lie_group_pose[1] * stheta) / lie_group_pose[2]
+        return [new_x, new_y, theta]
+
+
+def apply_affine_transformation(pose: Tuple[float, float, float],
+                                transformation: Tuple[float, float, float]) -> Tuple[float, float, float]:
+    """
+    Apply affine transformation to pose.
+    :param pose: Starting pose.
+    :param transformation: Affine transformation represented as a pose tuple.
+    :return: Pose tuple - the result of applying the transformation to the starting pose.
+    """
+
+    new_x = math.cos(pose[2]) * transformation[0] - math.sin(pose[2]) * transformation[1] + pose[0]
+    new_y = math.sin(pose[2]) * transformation[0] + math.cos(pose[2]) * transformation[1] + pose[1]
+    new_yaw = principal_value(pose[2] + transformation[2])
+
+    return [new_x, new_y, new_yaw]
+
+
+def get_lie_algebra(segment_sign: Tuple[int, int, int],
```

Comment:
In the comments Lie should be capitalized. Might also want to mark this as "private" with a leading underscore. I can't imagine we want anyone directly calling this function for any reason.

Reply:
Done!